### ISSUE TITLE
how to use this exporter

### ISSUE THREAD

**Issue by 13567436138** [oliver006/redis_exporter#174]
do i need to startup on instance for every redis node in the redis cluster

**Comment by ccs018** [comment]
No, just one instance of the redis exporter for your redis cluster.  In my case, I'm standing up multiple redis clusters and I have one redis exporter for each cluster.

**Comment by oliver006** [comment]
Interesting. @ccs018 - question: do you scrape all redis instances of the cluster with the one redis_exporter? Any advantages over one exporter per instance?

**Comment by ccs018** [comment]
> do you scrape all redis instances of the cluster with the one redis_exporter?

Yes.

> Any advantages over one exporter per instance?

Smaller footprint.  And I assumed this was what you intended since redis_exporter can be configured to scrape from multiple instances.

We have an on-prem product that uses multiple redis clusters with the expectation that the number of redis clusters will grow over time.  So I've developed a solution for both Fleet and Kubernetes to enable easy instantiations of redis clusters (with varying sizes and redis server configurations as needed). This "cookie-cutter" solution enables a lot of re-use and includes deploying a single redis_exporter per redis cluster and provides common set of grafana dashboards. This enables our application teams to focus on their applications and not worry too much about how to spin up a redis cluster. It also includes TLS between the redis nodes, redis exporter and redis clients (as part of a protection-in-depth security solution) along with client hostname validation to enable both authentication and authorization (restricts access to a particular redis cluster to only certain clients).

**Comment by oliver006** [comment]
> this was what you intended

Yeah, originally I did, for pretty much the same reasons (smaller footprint, mostly static setup).

Only later I learned that the more prometheus-y way of doing things is to let the Prometheus server handle any sort of service discovery and pair an exporter with every instance of what's being monitored/exported.

Anyway, it sounds like you found a setup that works for you and let's your developers focus on applications so that's good.

Thanks for sharing!
